android/expoview/src/main/java/versioned/host/exp/exponent/modules/universal/ScopedFirebaseCoreService.java

```diff
@@ -181,4 +205,26 @@ private static FirebaseOptions getOptionsFromManifest(JSONObject manifest) {
       return null;
     }
   }
+
+  // Registry lifecycle events
+
+  @Override
+  public void onCreate(ModuleRegistry moduleRegistry) {
+    // nop
+  }
+
+  @Override
+  public void onDestroy() {
+
+    // Mark this Firebase App as deleted. Don't delete it straight
+    // away, but mark it for deletion. When loading a new project
+    // a check is performed that will cleanup the deleted Firebase apps.
+    // This ensures that Firebase Apps don't get deleted/recreated
```

Comment:
i'm not totally clear on the implications of this but we may want to revisit this in the future if/when we add scoped native firebase modules in the client to give each reload a clean slate similar to if you re-open a standalone app. i don't think it matters much right now though so i'm ok with this

Reply:
That's correct. This is an optimisation, but also aimed at performance and safety. The safety part would be that there are no unnecessary tear-downs that could affect the app, e.g. when it is closed down. I do agree that this may mask underlying problems in case they exist. At the moment, the native project specific Firebase app is not really used indeed, so we can revisit this later when we have actual native services running in the Expo client.